src/west/main.py

```diff
@@ -112,21 +114,95 @@ def load_manifest(self):
+                log.wrn(mve_msg(self.mle, suggest_upgrade=False) +
+                        '\n  Cannot get extension command help, ' +
+                        "and most commands won't run." +
+                        '\n  To silence this warning, upgrade west.')
+                return
+            elif args.command in ['config', 'topdir']:
+                # config and topdir are safe to run, but let's
+                # warn the user that most other commands won't be.
+                log.wrn(mve_msg(self.mle, suggest_upgrade=False) +
+                        "\n  This should work, but most commands won't." +
+                        '\n  To silence this warning, upgrade west.')
+                return
+            elif args.command == 'init':
+                # init is fine to run -- it will print its own error,
+                # with context about where the installation was found,
+                # and what the user's choices are.
+                return
+            else:
+                assert args.command not in no_manifest_ok
+                log.die(mve_msg(self.mle))
+
+        # Other errors generally just fall back on no_manifest_ok.
+        def isinst(*args):
+            return any(isinstance(self.mle, t) for t in args)
+
+        if args.command not in no_manifest_ok:
+            if isinst(MalformedManifest, MalformedConfig):
+                log.die('\n  '.join(["can't load west manifest"] +
+                                    list(self.mle.args)))
+            elif isinst(FileNotFoundError):
+                # This should ordinarily only happen when the top
+                # level west.yml is not found.
+                log.die(f"file not found: {self.mle.filename}")
+            elif isinst(ManifestImportFailed):
+                if args.command == 'update':
+                    return      # that's fine
+
+                p, f = self.mle.project, self.mle.filename
+                ctxt = f'  Missing file: "{f}"'
+                if not isinstance(p, ManifestProject):
+                    # Try to be more helpful by explaining exactly
+                    # what west.manifest needs to happen before we can
+                    # resolve the missing import.
+                    rev = p.revision
+
+                    ctxt += f' from revision "{rev}"\n'
+                    ctxt += '  Hint: for this to work:\n'
+                    ctxt += f'          - {p.name} must be cloned\n'
+                    ctxt += (f'          - its {MANIFEST_REV_BRANCH} ref '
+                             'must point to a commit with the missing file\n')
+                    ctxt += (f'  Hint: to make sure {p.name} '
+                             f'is cloned and fetch {rev} from:\n')
+                    ctxt += f'          {p.url}\n'
+                    ctxt += (f'        then point {MANIFEST_REV_BRANCH} '
+                             'there, run:\n')
+                    ctxt += '          west update'
```

Comment:
Better but maybe a bit verbose:
```
FATAL ERROR: failed manifest import in zephyr (zephyr)
  Missing file: "west.yml" from revision "2c43bc81ea7b1de5332b806d99a718148375c981"
  Hint: for this to work:
          - zephyr must be cloned
          - its manifest-rev ref must point to a commit with the missing file
  Hint: to make sure zephyr is cloned and fetch 2c43bc81ea7b1de5332b806d99a718148375c981 from:
          https://github.com/NordicPlayground/fw-nrfconnect-zephyr
        then point manifest-rev there, run:
          west update
```

I think this is enough for users to get going and understand what was wrong:
```
FATAL ERROR: failed manifest import in zephyr (zephyr)
  Missing file: "west.yml" from revision "2c43bc81ea7b1de5332b806d99a718148375c981"
  Hint: for this to work:
          - zephyr must be cloned
          - its manifest-rev ref must point to a commit with the missing file
       To fix, run:
          west update
```

Reply:
In trying that out, I found that won't always fix it, and we may have bigger problems. If manifest-rev points at the commit, but that commit doesn't have that file, then `git ls-tree SHA nonexistent-commit` just prints nothing, causing west to crash. Will fix.

Reply:
> Will fix.

My initial idea for a fix didn't work out. I'll have to spend more time on this tomorrow.

Reply:
OK, I fixed it well enough for now. In true American cowboy fashion, I'm going to merge this and go on vacation for 2 weeks if CI passes.

Reply:
you mean commit and run style :laughing: